Code/Framework/AzCore/AzCore/Debug/Budget.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "Budget.h"
+
+#include <AzCore/Module/Environment.h>
+#include <AzCore/Math/Crc.h>
+#include <AzCore/Memory/SystemAllocator.h>
+
+AZ_DEFINE_BUDGET(Animation);
+AZ_DEFINE_BUDGET(Audio);
+AZ_DEFINE_BUDGET(AzCore);
+AZ_DEFINE_BUDGET(Editor);
+AZ_DEFINE_BUDGET(Entity);
+AZ_DEFINE_BUDGET(Game);
+AZ_DEFINE_BUDGET(System);
+AZ_DEFINE_BUDGET(Physics);
+
+namespace AZ::Debug
+{
+    struct BudgetImpl
+    {
+        AZ_CLASS_ALLOCATOR(BudgetImpl, AZ::SystemAllocator, 0);
+        // TODO: Budget implementation for tracking budget wall time per-core, memory, etc.
+    };
+
+    Budget::Budget(const char* name, uint32_t crc)
+        : m_name{ name }
+        , m_crc{ crc }
```

Comment:
Maybe better to just calculate the crc of the name here instead of relying on the calling code to do it? Doesn't really make much difference I guess, but would simplify it if anyone ever needed to construct one of these outside the context of the macros.

Reply:
Ah that's actually how it was done originally, but it was suggested to use `AZ_CRC_CE` which does this at compile time (this constructor is only every called from the `AZ_DEFINE_BUDGET` macro. Does that help or do you have another suggestion?

Reply:
All good, not a big deal.

Maybe we could add an additional explicit constructor that just takes the name and inits the crc there, just in case someone ever wants to use the simplified interface?